Steps returning data should be able to handle dict_values as well as list of rows

FAILED tests/test_builtin_steps.py::test_drop_no_dupes - phaser.exceptions.PhaserError: Step <function drop_duplicate_rows.<locals>.drop_duplicate_rows_step at 0x104832480> returned a <class 'dict_values'> rather than a list of rows

this would be not uncommon if somebody indexes the data as they do a batch step and then try to return the data from the index as I was doing

--- a/phaser/steps.py
+++ b/phaser/steps.py
@@ -150,6 +150,9 @@ def batch_step(func=None, *, extra_sources=None, extra_outputs=None, check_size=
         raise exc
 
     def postprocess(step_function, result):
+        if isinstance(result, type({}.values())):
+            # If a step returned the values of a dict and forgot to convert to a list, we can convert for it.
+            result = list(result)
         if not isinstance(result, Sequence):
             raise PhaserError(
                 f"Step {step_function} returned a {result.__class__} rather than a list of rows")

--- a/tests/test_steps.py
+++ b/tests/test_steps.py
@@ -141,6 +141,19 @@ def test_simple_batch_step():
     assert phase.row_data == [{'value': 1, 'sum': 1}, {'value': 2, 'sum': 3}]
 
 
+def test_batch_step_returns_dict_array():
+    @batch_step
+    def return_as_dict_array(batch, context):
+        the_dict = {}
+        for (index, row) in enumerate(batch):
+            the_dict[index] = row
+        return the_dict.values()
+
+    phase = Phase(steps=[return_as_dict_array])
+    phase.load_data([{'id': 1, 'val': 'a'}])
+    phase.run_steps()
+    assert phase.row_data == [{'id': 1, 'val': 'a'}]
+
 def test_batch_step_cant_use_drop_row_exception():
     @batch_step
     def try_something_nonsensical(batch, context):

